### PR TITLE
backupccl: display up to 10 missing files in `SHOW BACKUP .. with check_files`

### PR DESCRIPTION
**Pull request by msbutler**
Previously, `SHOW BACKUP WITH check_files` displayed the first missing SST.
This patch will display up to 100 missing SSTs. Further, this renames the
misleading `approximateTablePhysicalSize` to `approximateSpanPhysicalSize`.
Below I write out how physical table size is calculated:

1. Each range we backup maps to 1 to many spans (currently in the
backup_manfest.files object).

2. 1 to many spans get written to an SST. No span will get written to multiple
SSTs.

3. When backup created these spans, it tried really hard to split spans at
table boundaries, so only one table’s data could be in a span, but a last
minute table creation makes this near impossible, due to slow range splits.
A big table will have many spans.

4. To compute the approximate logical size (called size_bytes in SHOW BACKUP)
of each table, we sum the logical bytes over all it’s spans. We identify a
table’s span by checking the table prefix of the first key in the span. See
getTableSizes method)

5. To compute the physical size (file_bytes in SHOW BACKUP) of a span, compute
the logical size of each SST by summing the logical bytes in the SST  over its
spans (see getLogicalSSTSize method), and attribute a portion of the physical
SST size (returned from cloud storage) to a span  using the formula:
(sstPhysicalSize) * (logicalSpanSize) / (logicalSSTSize) = physicalSpanSize (
the approximateSpanTableSize method implements this).

6. To compute the physical size of a table, sum over the physical sizes the
table’s spans

This patch cleans up code from #80491

Release note (sql change): SHOW BACKUP WITH check_files will display up to 10
missing SSTs.

### CODE REVIEW

File: pkg/ccl/backupccl/show.go [hunk 1]
```diff
@@ -486,6 +486,9 @@ func checkBackupFiles(
 	storeFactory cloud.ExternalStorageFromURIFactory,
 	user username.SQLUsername,
 ) ([][]int64, error) {
+	maxMissingFiles := 10
```
Comment (dt) [inline pkg/ccl/backupccl/show.go:489]: nit: const maxMissingFiles = 10

File: pkg/ccl/backupccl/show.go [hunk 2]
```diff
@@ -486,6 +486,9 @@ func checkBackupFiles(
...
+	missingFiles := make(map[string]bool, maxMissingFiles)
+	numMissingFiles := 0
```
Comment (dt) [inline pkg/ccl/backupccl/show.go:491]: I think you can just use len() on the map?

Comment (msbutler) [reply pkg/ccl/backupccl/show.go]: Done.

File: pkg/ccl/backupccl/show.go [hunk 3]
```diff
@@ -559,8 +572,24 @@ func checkBackupFiles(
...
 		manifestFileSizes[layer] = layerFileSizes
 	}
+	if numMissingFiles > 0 {
+		filesForMsg := make([]string, numMissingFiles)
+		i := 0
+		for file := range missingFiles {
+			filesForMsg[i] = file
+			i++
+		}
+		errorMsgPrefix := "The following files are missing from the backup:"
```
[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/ccl/backupccl/show.go:584]: Done

**Review by dt — APPROVED** [review]

**Comment by msbutler** [comment]
bors r=dt

**Pull request merged**